cartography/data/jobs/analysis/aws_ec2_keypair_analysis.json

```diff
@@ -0,0 +1,30 @@
+{
+    "name": "Analysis jobs for EC2 Key Pairs",
+    "statements": [
+        {
+            "__comment__": "Delete the attribute uploaded",
+            "query": "MATCH (k:EC2KeyPair) WHERE EXISTS (k.uploaded) REMOVE k.uploaded return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "Delete the attribute duplicate_keyfingerprint",
+            "query": "MATCH (k:EC2KeyPair) WHERE EXISTS (k.duplicate_keyfingerprint) REMOVE k.duplicate_keyfingerprint return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "Delete stale relationships between EC2KeyPairs",
+            "query": "MATCH (k1:EC2KeyPair)-[r:MATCHING_FINGERPRINT]-(k2:EC2KeyPair) WHERE r.lastupdated <> {UPDATE_TAG} DELETE (r) return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "Set EC2KeyPairs with MD5 hash keyfingerprints (47 characters, instead of 59 characters) as uploaded = True",
+            "query": "MATCH (k:EC2KeyPair) WHERE length(k.keyfingerprint) = 47 SET k.uploaded = True return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "Attach EC2KeyPairs with matching fingerprints to eachother and set duplicate_keyfingerprint = True",
+            "query": "MATCH (k1:EC2KeyPair), (k2:EC2KeyPair) WHERE k1.id <> k2.id AND k1.keyfingerprint = k2.keyfingerprint SET k1.duplicate_keyfingerprint = True, k2.duplicate_keyfingerprint = True MERGE (k1)-[r:MATCHING_FINGERPRINT]-(k2) ON CREATE SET r.firstseen = {UPDATE_TAG} SET r.lastupdated = {UPDATE_TAG} return COUNT(*) as TotalCompleted",
```

Comment:
Ah I think I see why you didn't add directionality - k1 matches with k2, and you can also say that k2 matches with k1.  Hm, let's think on this a bit more if having a single direction is possible (or beneficial).

Reply:
Yeah, Neo4j [will add directionality, anyways](https://neo4j.com/docs/cypher-manual/current/clauses/merge/#merge-merge-on-an-undirected-relationship).